decidim-core/app/controllers/decidim/locales_controller.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+require_dependency "decidim/application_controller"
+
+module Decidim
+  # A controller to allow users switching their locale.
+  class LocalesController < ApplicationController
+    def create
```

Comment:
Well, it makes sense that the user's state can't be changed by a potentially malitious user sending a link. Ok, I'm cool with this.